code/modules/events/event.dm

```diff
@@ -91,6 +91,7 @@
   *
   * Allows you to start before announcing or vice versa.
   * Only called once.
+  * Ensure no sleep is called. Use INVOKE_ASYNC to call procs which do.
```

Comment:
It literally says here `Only called once`.

Reply:
![image](https://user-images.githubusercontent.com/15887760/103220405-7bd0e180-4920-11eb-9450-8828d5ecbda0.png)
Any of the calls there can be called multiple times if the first processing call gets put to sleep long enough for another one to spawn before line 167 is hit

Reply:
Then make the code actually work properly, and put it in a correct order, such as

`Start > Announce > Tick > End > Kill`

Reply:
It doesn't have to be done in that order is the thing. The announce can happen before the start. The end can happen before the announce and such. Start should happen before the end ofcourse

Reply:
You are still trying to beat around the point of refusing to add actual protection into avoiding things starting twice, by forcing every event proc to be invoked async, which is not a good idea. BYOND is a synchronous language. At this point you may as well just invoke the actual start async instead of having all procs inside have to be snowflaked. Or just add an actual check to make sure it doesnt start 3 times, like I have asked you to do 3 times now.

Reply:
I've listed my reasoning to not do that earlier. It will make ticks between events be irregular. One time the time between can be 2 seconds the other time 6 which makes the system unpredictable. Having the system work like in the PR will not have that effect and instead the event will just continue on as expected. Feel free to make the PR yourself to change it the way you want.